daemon/containerd/image_push.go

```diff
@@ -134,24 +134,22 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 		return err
 	}
 
-	addChildrenToJobs := containerdimages.HandlerFunc(
+	addLayerJobs := containerdimages.HandlerFunc(
 		func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
-			children, err := containerdimages.Children(ctx, store, desc)
-			if err != nil {
-				return nil, err
-			}
-			for _, c := range children {
-				jobsQueue.Add(c)
+			switch {
```

Comment:
Why not a simple `if containerdimages.IsLayerType(desc.MediaType) { ... `

Reply:
That would exclude other potentially big blobs that are not explicitly layers.
Currently this would exclude buildkit attestations with media type `application/vnd.in-toto+json`.